### DISCUSSION TITLE
image display home screen  budibase

### DISCUSSION THREAD

**Discussion by vereng** [Budibase/budibase#10571] [Help]
created successfuly a table of image attachments (column names: image_name,image_file)
2 under home screen under header added component data_provider named images
pointing to the images_table
3 added repeater component
4 added image component with image name like "lotfi1" exists as one of the attachments in table

5 inserted the binding field text pointing to the image file within the table
      according to instructions eamples
                                             as follows:
{{url.images Repeater.images_table.image_file}} not work
{{images Repeater.images_table.image_file.0}} not work
{{url.images Repeater.images.image_file}} not work
{{images Repeater.images.image_file}} not work
{{images_table Repeater.images.image_file}} not work
{{images Repeater.images_table.image_file[0].0}} not work

6 expected behaviour
when previewing the screen created the file name "image1" may appear but not the already uploaded image from table

image
![image](https://github.com/Budibase/budibase/assets/20261976/7c1aa1fb-d342-4cd8-8588-536be0cfb3b1)

**Comment by melohagan** [comment]
Hey @vereng 

Docs: https://docs.budibase.com/docs/image#displaying-images-from-an-attachment

There are two options. One is to loop over the rows in your table:

> images (Data provider)
> > repeater
> > > image attachment field (Data provider)
> > > > repeater
> > > > > image component bound to attachment image URL

There's a screenshot in the docs page showing the above. You can also short-hand that using **Repeater blocks**, which would condense that into:

> images (Repeater block)
> > image attachment field (Repeater block)
> > > image component bound to attachment image URL

Alternatively, if you know that your *images* table has an attachment column with a single image (and never any more than one image), then you can skip one of the repeaters:

> images (Data provider)
> > repeater
> > > image component bound to: `{{ Repeater.images.AttachmentFieldContainingASingleImage.0.url }}`

**Reply by melohagan** [reply]
Also I appreciate that may not be entirely intuitive, and an issue has been raised around making this a little easier: https://github.com/Budibase/budibase/issues/8807
